dodo.py

```diff
@@ -64,115 +114,128 @@ def task_po():
-                    ]
-                )
-                print("\n", result)
-                # print("stdout:", result.stdout)
+            LOC_PATH = os.path.join("tabcmd", "locales", current_locale, "LC_MESSAGES")
+            PROPS_FILE = os.path.join(LOC_PATH, "transcoded.properties")
+            PO_FILE = os.path.join(LOC_PATH, "tabcmd.po")
+            LOG_FILE = os.path.join(LOC_PATH, "prop2po.out")
+            with open(LOG_FILE, "w+", encoding="utf-8") as logfile:
+                try:
+                    result = subprocess.run(
+                        [
+                            "python",
+                            "bin/i18n/prop2po.py",
+                            "--encoding",
+                            "utf-8",  # for the .po header
+                            "--language",
+                            current_locale,  # for the .po header
+                            "--project",
+                            "Tabcmd 2",
+                            "--copyright",
+                            "TABLEAU SOFTWARE, LLC, A SALESFORCE COMPANY. ALL RIGHTS RESERVED",
+                            PROPS_FILE,
+                            PO_FILE
+                        ], 
+                        stdout=logfile,
+                        stderr=logfile
+                    )
+                    print("Written from {} to {}".format(PROPS_FILE, PO_FILE))
+                except Exception as e:
+                    print("run for {} failed with exception".format(current_locale))
+                    print("see log file {}".format(LOG_FILE))
+                    exit(1)
+        
                 if not result.returncode == 0:
-                    print("stderr:", result.stderr)
+                    print("FAILED")
+                    print("see log file {}".format(LOG_FILE))
+                    exit(1)
+        print("Done")            
 
     return {
         "actions": [process_locales],
         "verbosity": 2,
     }
 
 
+# clean: all we need to keep are the provided translation.properties files from the monolith, at locales/[current_locale]
+# and the final tabcmd.mo files in LC_MESSAGES generated by 
+# >doit properties po mo
 def task_clean_all():
-    """For all languages: removes all generated artifacts (.po, .mo) which source from properties files."""
+    """For all languages: removes all generated intermediate files (properties, po) from the loc build."""
 
     def process_locales():
         for current_locale in LOCALES:
-            LOC_PATH = "tabcmd/locales/" + current_locale
-            for file in glob.glob(LOC_PATH + "/*.properties"):
-                basename = os.path.basename(file).split(".")[0]
-                print("deleting", basename + ".*")
-                try:
-                    os.remove(LOC_PATH + "/LC_MESSAGES/" + basename + ".po")
-                except OSError:
-                    pass
+            LOC_PATH = os.path.join("tabcmd", "locales", current_locale, "LC_MESSAGES", "*.p*")
```

Comment:
Feels a little scary, like the glob expression couldn't accidentally match a file type in the future we don't want it to. Consider being explicit and use one of the approaches in this thread: https://stackoverflow.com/a/4568638

Reply:
updated this to use specific file extensions